File: apps/frontend/src/types/post.ts
/**
 * 投稿型
 */
export interface Post {
  /** 投稿ID */
  id: string
  /** 投稿本文 */
  content: string
  /** 投稿日時 */
  createdAt: string
}

/**
 * 投稿作成リクエスト型
 */
export interface PostCreateRequest {
  content: string
}

File: apps/backend/src/main/kotlin/io/github/yna87/vuekotlinsns/entity/Post.kt
package io.github.yna87.vuekotlinsns.entity

import java.time.OffsetDateTime
import java.util.UUID

/**
 * 投稿エンティティ
 */
data class Post(
    val id: UUID,
    val userId: UUID,
    val content: String,
    val createdAt: OffsetDateTime,
    val user: User,
)
